kamel run runtime/examples/Sample.java fails with java.lang.NoClassDefFoundError

Exception in thread "main" org.joor.ReflectException: java.lang.reflect.InvocationTargetException
--
  | at org.joor.Reflect.on(Reflect.java:781)
  | at org.joor.Reflect.call(Reflect.java:463)
  | at org.joor.Compile.compile(Compile.java:73)
  | at org.joor.Reflect.compile(Reflect.java:77)
  | at org.apache.camel.k.jvm.RoutesLoaders$2.load(RoutesLoaders.java:90)
  | at org.apache.camel.k.jvm.Application.main(Application.java:34)
  | Caused by: java.lang.reflect.InvocationTargetException
  | at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  | at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  | at java.lang.reflect.Method.invoke(Method.java:498)
  | at org.joor.Reflect.on(Reflect.java:777)
  | ... 5 more
  | Caused by: java.lang.NoClassDefFoundError: etc/camel/Sample (wrong name: Sample)
  | at java.lang.ClassLoader.defineClass1(Native Method)
  | at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
  | at java.lang.ClassLoader.defineClass(ClassLoader.java:642)
  | ... 10 more

--- a/runtime/jvm/src/main/java/org/apache/camel/k/jvm/RoutesLoaders.java
+++ b/runtime/jvm/src/main/java/org/apache/camel/k/jvm/RoutesLoaders.java
@@ -19,6 +19,7 @@ package org.apache.camel.k.jvm;
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.io.Reader;
+import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
 import java.util.List;
 import java.util.function.Function;
@@ -85,9 +86,12 @@ public enum RoutesLoaders implements RoutesLoader {
             try (InputStream is = Routes.loadResourceAsInputStream(resource)) {
                 String name = StringUtils.substringAfter(resource, ":");
                 name = StringUtils.removeEnd(name, ".java");
-                name = StringUtils.removeStart(name, "/");
 
-                return Reflect.compile(name.replace("/", "."), IOUtils.toString(is)).create().get();
+                if (name.contains("/")) {
+                    name = StringUtils.substringAfterLast(name, "/");
+                }
+                
+                return Reflect.compile(name, IOUtils.toString(is, StandardCharsets.UTF_8)).create().get();
             }
         }
     },

